### ISSUE TITLE
docker_container module error in ansible 2.6.1

### ISSUE THREAD

**Issue by sagarpinninti** [ansible/ansible#43297]
##### SUMMARY

While trying to start the container from centos:7 image i am getting module_stderr"
##### ISSUE TYPE
 - Bug Report

##### ANSIBLE VERSION

```
ansible 2.6.1
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/root/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 2.7.5 (default, May 31 2018, 09:41:32) [GCC 4.8.5 20150623 (Red Hat 4.8.5-28)]

```

##### OS / ENVIRONMENT

RHEL7.5
##### STEPS TO REPRODUCE

```yaml
-bash-4.2$ cat test1.yml
---
-bash-4.2$ cat test1.yml
---
- hosts: localhost
  tasks:
  - name: Container present
    docker_container:
     name: mycontainer
     state: present
     image: centos:7
     command: sleep infinity



```

##### EXPECTED RESULTS

container started
##### ACTUAL RESULTS

```
task path: /var/lib/jenkins/workspace/ans_con/test1.yml:4
<127.0.0.1> ESTABLISH LOCAL CONNECTION FOR USER: jenkins
<127.0.0.1> EXEC /bin/sh -c 'echo ~jenkins && sleep 0'
<127.0.0.1> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /var/lib/jenkins/.ansible/tmp/ansible-tmp-1532589376.31-71984925192585 `" && echo ansible-tmp-1532589376.31-71984925192585="` echo /var/lib/jenkins/.ansible/tmp/ansible-tmp-1532589376.31-71984925192585 `" ) && sleep 0'
Using module file /usr/lib/python2.7/site-packages/ansible/modules/cloud/docker/docker_container.py
<127.0.0.1> PUT /var/lib/jenkins/.ansible/tmp/ansible-local-25893ZjaZ_g/tmp1pZX61 TO /var/lib/jenkins/.ansible/tmp/ansible-tmp-1532589376.31-71984925192585/docker_container.py
<127.0.0.1> EXEC /bin/sh -c 'chmod u+x /var/lib/jenkins/.ansible/tmp/ansible-tmp-1532589376.31-71984925192585/ /var/lib/jenkins/.ansible/tmp/ansible-tmp-1532589376.31-71984925192585/docker_container.py && sleep 0'
<127.0.0.1> EXEC /bin/sh -c 'sudo -H -S -n -u root /bin/sh -c '"'"'echo BECOME-SUCCESS-fhcmhrggodzodvgujajspzjlhsayjiwb; /usr/bin/python2 /var/lib/jenkins/.ansible/tmp/ansible-tmp-1532589376.31-71984925192585/docker_container.py'"'"' && sleep 0'
<127.0.0.1> EXEC /bin/sh -c 'rm -f -r /var/lib/jenkins/.ansible/tmp/ansible-tmp-1532589376.31-71984925192585/ > /dev/null 2>&1 && sleep 0'
The full traceback is:
Traceback (most recent call last):
  File "/tmp/ansible_7piE7K/ansible_module_docker_container.py", line 2081, in <module>
    main()
  File "/tmp/ansible_7piE7K/ansible_module_docker_container.py", line 2076, in main
    cm = ContainerManager(client)
  File "/tmp/ansible_7piE7K/ansible_module_docker_container.py", line 1703, in __init__
    self.present(state)
  File "/tmp/ansible_7piE7K/ansible_module_docker_container.py", line 1723, in present
    new_container = self.container_create(self.parameters.image, self.parameters.create_parameters)
  File "/tmp/ansible_7piE7K/ansible_module_docker_container.py", line 825, in create_parameters
    host_config=self._host_config(),
  File "/tmp/ansible_7piE7K/ansible_module_docker_container.py", line 931, in _host_config
    return self.client.create_host_config(**params)
  File "/usr/lib/python2.7/site-packages/docker/api/container.py", line 157, in create_host_config
    return utils.create_host_config(*args, **kwargs)
TypeError: create_host_config() got an unexpected keyword argument 'init'

fatal: [localhost]: FAILED! => {
    "changed": false,
    "module_stderr": "Traceback (most recent call last):\n  File \"/tmp/ansible_7piE7K/ansible_module_docker_container.py\", line 2081, in <module>\n    main()\n  File \"/tmp/ansible_7piE7K/ansible_module_docker_container.py\", line 2076, in main\n    cm = ContainerManager(client)\n  File \"/tmp/ansible_7piE7K/ansible_module_docker_container.py\", line 1703, in __init__\n    self.present(state)\n  File \"/tmp/ansible_7piE7K/ansible_module_docker_container.py\", line 1723, in present\n    new_container = self.container_create(self.parameters.image, self.parameters.create_parameters)\n  File \"/tmp/ansible_7piE7K/ansible_module_docker_container.py\", line 825, in create_parameters\n    host_config=self._host_config(),\n  File \"/tmp/ansible_7piE7K/ansible_module_docker_container.py\", line 931, in _host_config\n    return self.client.create_host_config(**params)\n  File \"/usr/lib/python2.7/site-packages/docker/api/container.py\", line 157, in create_host_config\n    return utils.create_host_config(*args, **kwargs)\nTypeError: create_host_config() got an unexpected keyword argument 'init'\n",
    "module_stdout": "",
    "msg": "MODULE FAILURE",
    "rc": 1
}
        to retry, use: --limit @/var/lib/jenkins/workspace/ans_con/test1.retry

PLAY RECAP **********************************************************************************************************************************************************************************************************************************
localhost                  : ok=1    changed=0    unreachable=0    failed=1
```

```

```

**Comment by joren485** [comment]
duplicate of #42162

**Comment by Akasurde** [comment]
@sagarpinninti Thanks for reporting this issue. Could you please check above issue and let us know if it works for you ?

**Comment by samdoran** [comment]
@sagarpinninti This definitely looks like an issue with the differing Docker libraries. Please trying removing the suggested packages and installing only `docker` or `docker-py`.

needs_info

**Comment by felixfontein** [comment]
For Ansible 2.6.1, installing only `docker-py` will not work. You need to remove `docker-py` and install `docker` (if it was already installed, do a force-reinstall or also first remove it and then install it again; otherwise it will probably be broken afterwards).

**Comment by felixfontein** [comment]
Ansible 2.6.2 was released today, which contains a fix for this issue.

**Comment by sagarpinninti** [comment]
Fixed in ansible 2.6.2

**Comment by sagarpinninti** [comment]
Fixed in ansible 2.6.2
